perfdash/metrics-downloader.go

```diff
@@ -158,7 +158,7 @@ func (g *Downloader) artifactName(jobAttrs Tests, file string) string {
 }
 
 func getResultCategory(metricsFileName string, filePrefix string, category string, artifacts []string) string {
-	if len(artifacts) <= 1 {
+	if len(artifacts) <= 1 || strings.HasPrefix(metricsFileName, GenericPrometheusQueryMeasurementName) {
```

Comment:
Please add a bool "enforceConstantCategory" or something like that to TestDescription, set it for GenericMeasurements and check it here.

This way, we will pattern match files only in a single place (and e.g. if we will change to some other model, e.g. metadata-based instead of file-name based) it will be easier to reason about changes to be made.